### PR TITLE
Add support for options in the CLI

### PR DESCRIPTION
**Pull request by joanrieu**
This PR adds support in the CLI for passing compilation options in the CLI.

For example, you can use flags like `--unreachableDefinitions` to change the behavior of the program.

NB: It's far from perfect since I didn't add validation of the arguments. You might want to add that, as well as tests, if you feel that it's needed.

### CODE REVIEW

**Comment by bcherny** [comment]
@joanrieu Thanks for the contribution! Once you add tests I'll be happy to merge this - this package has enough users that tests are really important for catching edge cases and regressions - see [CONTRIBUTING.md](https://github.com/bcherny/json-schema-to-typescript/blob/master/CONTRIBUTING.md).

Please also add one or two examples to the README. For example, how do I pass a `style` option like `singleQuote` in?

**Comment by joanrieu** [comment]
@bcherny Fair enough, I'll see if I can find the time, thanks.

**Comment by joanrieu** [comment]
@bcherny I added tests and improved the code and examples, you should be able to merge it now.

The Ava snapshots are apparently out-of-date on `master` so I didn't update them on my branch by fear of breaking things. Would you mind taking care of that? 👍

File: src/cli.ts [hunk 1]
```diff
@@ -61,7 +61,7 @@ function printHelp() {
   process.stdout.write(
 `
 ${pkg.name} ${pkg.version}
-Usage: json2ts [--input, -i] [IN_FILE] [--output, -o] [OUT_FILE]
+Usage: json2ts [--input, -i] [IN_FILE] [--output, -o] [OUT_FILE] [OPTIONS]
```
Comment (bcherny) [inline src/cli.ts]: Please add a line explaining what OPTIONS is (similar to what you did in the README)

File: test/testCLI.ts [hunk 2]
```diff
@@ -34,6 +34,18 @@ export function run() {
...
 
+  test('file in (-i), unreachable definitions flag, pipe out', t => {
+    t.snapshot(
+      execSync('node dist/src/cli.js -i ./test/resources/DefinitionsOnly.json --unreachableDefinitions').toString()
+    )
+  })
+
+  test('file in (-i), style flags, pipe out', t => {
+    t.snapshot(
+      execSync('node dist/src/cli.js -i ./test/resources/DefinitionsOnly.json --style.singleQuote --no-style.semi').toString()
```
Comment (bcherny) [inline test/testCLI.ts]: Please modify `DefinitionsOnly.json` so you're actually testing `style.singleQuote`.

**Comment by joanrieu** [comment]
@bcherny Done and done, and I also committed the updated test snapshots. All good now?

As a side note, I found out what happened regarding the snapshots: Ava generates snapshots in a different order with `ava --update-snapshots` instead of `npm test` (which runs `ava --verbose`), which made me think the ones on master were not up to date (git reported a lot of changes, but the contents are the same except, only in a different order).

File: src/cli.ts (hunk 1 above)
Comment (joanrieu) [reply src/cli.ts]: GitHub didn't close this review comment but the docs have actually been added to the help text below that line.

**Comment by bcherny** [comment]
@joanrieu Sorry, I didn't see your updates. Merged and published - thanks for the contribution!

```
+ json-schema-to-typescript@5.3.0
```
